examples/deploy_tasks.yaml

```diff
@@ -36,44 +36,44 @@ metadata:
   name: deploy-with-kubectl
   namespace: default
 spec:
-    inputs:
-        resources:
-            - name: workspace
-              type: git
-            - name: redisImage
-              type: image
-            - name: guestbookImage
-              type: image
-        params:
-            - name: kubectlArgs
-              description: Extra arguments to pass to kubectl
-            - name: pathToFiles
-              description: Path to the manifests to apply
-        clusters:
-            - name: targetCluster
-    buildSpec:
-        steps:
-        - name: replaceRedisImage
-          image: busybox
-          command: ['sed']
-          args:
-          - "-ri"
-          - "'s/image: k8s.gcr.io\\/redis:e2e/image: ${inputs.resources.redisImage.url}@{inputs.resources.redisImage.digest}/' ${inputs.params.pathToFiles}"
-        - name: replaceGuestbookImage
-          image: busybox
-          command: ['sed']
-          args:
-          - "-ri"
-          - "'s/image: gcr.io\\/google-samples\\/gb-frontend:v4/image: ${inputs.resources.guestbookImage.url}@{inputs.resources.guestbookImage.digest}/' ${inputs.params.pathToFiles}"
-        - name: runKubectl
-          image: lachlanevenson/k8s-kubectl
-          command: ['kubectl']
-          args:
-          - '--use-context'
-          - '${inputs.clusters.targetCluster}'
-          - '--namespace'
-          - '${inputs.clusters.targetCluster.namespace}'
-          - 'apply'
-          - '-f'
-          - '${inputs.params.pathToFiles}'
-          - '${inputs.params.kubectlArgs}'
+  inputs:
+    resources:
+    - name: workspace
+      type: git
+    - name: redisImage
+      type: image
+    - name: guestbookImage
+      type: image
+    - name: targetCluster
+      type: cluster  
+    params:
+    - name: kubectlArgs
+      description: Extra arguments to pass to kubectl
+    - name: pathToFiles
+      description: Path to the manifests to apply
+  buildSpec:
+    steps:
+    - name: replaceRedisImage
+      image: busybox
+      command: ['sed']
+      args:
+      - "-ri"
+      - "'s/image: k8s.gcr.io\\/redis:e2e/image: ${inputs.resources.redisImage.url}@{inputs.resources.redisImage.digest}/' ${inputs.params.pathToFiles}"
+    - name: replaceGuestbookImage
+      image: busybox
+      command: ['sed']
+      args:
+      - "-ri"
+      - "'s/image: gcr.io\\/google-samples\\/gb-frontend:v4/image: ${inputs.resources.guestbookImage.url}@{inputs.resources.guestbookImage.digest}/' ${inputs.params.pathToFiles}"
+    - name: runKubectl
+      image: lachlanevenson/k8s-kubectl
+      command: ['kubectl']
+      args:
+      - '--use-context'
+      - '${inputs.resource.targetCluster.clusterName}'
```

Comment:
similarly here i think we'd need to give kubectl the path to the kubeconfig file